zap/src/main/java/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -209,33 +211,66 @@ public void setMessage(HttpMessage msg) {
     @Override
     public String setParameter(
             HttpMessage msg, NameValuePair originalPair, String name, String value) {
-        return setParameter(msg, originalPair, value);
+        return setParameter(msg, Arrays.asList(originalPair), Arrays.asList(value));
     }
 
     @Override
     public String setEscapedParameter(
             HttpMessage msg, NameValuePair originalPair, String name, String value) {
-        return setParameter(msg, originalPair, value);
+        return setParameter(msg, Arrays.asList(originalPair), Arrays.asList(value));
     }
 
-    private String setParameter(HttpMessage msg, NameValuePair originalPair, String value) {
+    @Override
+    public List<String> setParameters(HttpMessage msg, List<AppParameter> appParameters) {
+        return Arrays.asList(
```

Comment:
As for single parameter setParameter method is returning HTTP message's body so returned the same(by adding to list). @thc202

Reply:
Well, that allows to provide evidence of what was injected for instance. Although for multiple parameters it would be harder to provide a more precise evidence I guess.